bfabric/bfabric_config.py

```diff
@@ -27,73 +30,146 @@ class BfabricConfig:
+    - If not, secondly, the parser will check if the environment variable `BFABRICPY_CONFIG_ENV` is declared
+    - If not, finally, the parser will select the default_config specified in [GENERAL] of the .bfabricpy.yml file 
+'''
+
+
+def _read_config_env_as_dict(config_path: str, config_env: str = None) -> Tuple[str, dict]:
+    """
+    Reads and partially parses a bfabricpy.yml file
+    :param config_path:  Path to the configuration file. It is assumed that it exists
+    :param config_env:   Specific environment to parse. If not provided, it is deduced from an environment variable
+       or the config file itself.
+    :return: Returns a target environment name, and the corresponding data from bfabricpy.yml file as a dictionary
+    """
+
     """Parses a .bfabricrc.py file and returns a tuple of BfabricConfig and BfabricAuth objects."""
-    values = {}
-    file_path = os.path.realpath(file.name)
     logger = logging.getLogger(__name__)
-    logger.info(f"Reading configuration from: {file_path}")
-
-    for line in file:
-        if line.startswith("#"):
-            continue
-
-        key, _, value = [part.strip() for part in line.partition("=")]
-        if key not in [
-            "_PASSWD",
-            "_LOGIN",
-            "_WEBBASE",
-            "_APPLICATION",
-            "_JOB_NOTIFICATION_EMAILS",
-        ]:
-            continue
-
-        # In case of multiple definitions, the first rule counts!
-        if key not in values:
-            if key in ["_APPLICATION"]:
-                try:
-                    values[key] = json.loads(value)
-                except json.JSONDecodeError as e:
-                    raise ValueError(
-                        f"While reading {file_path}. '{key}' is not a valid JSON string."
-                    ) from e
-            else:
-                # to make it downward compatible; so we replace quotes in login and password
-                values[key] = value.replace('"', "").replace("'", "")
+    logger.info(f"Reading configuration from: {config_path}")
+
+    # Read the config file
+    config_dict = yaml.safe_load(Path(config_path).read_text())
+
+    # config = ConfigParser()
+    # config.read(config_path)
+    # config_dict = {s: dict(config.items(s)) for s in config.sections()}
+
+    if "GENERAL" not in config_dict:
+        raise IOError("Config file must have a general section")
+    if 'default_config' not in config_dict['GENERAL']:
+        raise IOError("Config file must provide a default environment")
+    config_env_default = config_dict['GENERAL']['default_config']
+
+    # Determine which environment we will use
+    # By default, use the one provided by config_env
+    if config_env is None:
+        # Try to find a relevant
+        config_env = os.getenv("BFABRICPY_CONFIG_ENV")
+        if config_env is None:
+            logger.log(20, "BFABRICPY_CONFIG_ENV not found, using default environment " + config_env_default)
```

Comment:
Level 20 is equivalent to `logging.INFO` but you can even use the shorthand here:

```suggestion
            logger.info(f"BFABRICPY_CONFIG_ENV not found, using default environment {config_env_default}")
```